File: mtool/reference/legacy-mtool-templates/dbclasses_template_system_default/swift/template-dataclass.swift

import Foundation

// == START OF EDITABLE AREA FOR ABOVE ==
// == END OF EDITABLE AREA FOR ABOVE ==

// THIS CLASS WAS AUTOMATICALLY CREATED BY MTOOL. DON'T EDIT EXCEPT THE EDITABLE AREA.
// [JP] このクラスは自動生成されています。編集可能領域以外は変更しないで下さい。

public class __CLASS_NAME____INHERIT_CLASS__
{
__AUTOMATED_CODE_COMES_HERE__

    override init()
    {
        // == START OF EDITABLE AREA FOR INIT ==
        // == END OF EDITABLE AREA FOR INIT ==
        super.init();
    }
    override init(json:Any?)
    {
        var dic:Dictionary<String, Any>? = json as? Dictionary<String, Any>;
        if (dic != nil) {
__DESERIALIZE__
            // == START OF EDITABLE AREA FOR INIT WITH DIC ==
            // == END OF EDITABLE AREA FOR INIT WITH DIC ==
        }
        super.init(json:json);
    }
    override func Serialize() -> Dictionary<String, Any>
    {
__SERIALIZE_DEFINE__
__SERIALIZE__
        // == START OF EDITABLE AREA FOR SERIALIZE ==
        // == END OF EDITABLE AREA FOR SERIALIZE ==
        return dic;
    }

// == START OF EDITABLE AREA FOR ADDITIONAL CLASS IMPLEMENTATION DEFINITION ==
// == END OF EDITABLE AREA FOR ADDITIONAL CLASS IMPLEMENTATION DEFINITION ==

}

__CONST_DEFINITION__

// == START OF EDITABLE AREA FOR BOTTOM ==
// == END OF EDITABLE AREA FOR BOTTOM ==

File: mtool/reference/legacy-mtool-templates/dbclasses_template_system_default/cs/template-dataclass.cs

// == START OF EDITABLE AREA FOR ABOVE ==
// == END OF EDITABLE AREA FOR ABOVE ==

namespace __CS_NAMESPACE__
{
    // THIS CLASS WAS AUTOMATICALLY CREATED BY MTOOL. DON'T EDIT EXCEPT THE EDITABLE AREA.
    // [JP] このクラスは自動生成されています。編集可能領域以外は変更しないで下さい。
    
    public class __CLASS_NAME____INHERIT_CLASS__System.ComponentModel.INotifyPropertyChanged
    {
        public __CLASS_NAME__()
        {
            // == START OF EDITABLE AREA FOR DEFAULT CONSTRUCTOR ==
            // == END OF EDITABLE AREA FOR DEFAULT CONSTRUCTOR ==
        }

__AUTOMATED_CODE_COMES_HERE__

        #region PropertyChanged
        public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;
        protected virtual void OnPropertyChanged(string name)
        {
            if (PropertyChanged == null) return;
            PropertyChanged(this, new System.ComponentModel.PropertyChangedEventArgs(name));
        }
        #endregion

        #region Shallow Clone
        public __CLASS_NAME__ CloneShallow()
        {
            return (__CLASS_NAME__)this.MemberwiseClone();
        }
        #endregion

        #region Copy Properties From
        internal void CopyPropertiesFrom(__CLASS_NAME__ source_obj)
        {
__COPY_FROM_VALUE__

__COPY_FROM_RAISE_EVENT__
        }
        #endregion

        // == START OF EDITABLE AREA FOR ADDITIONAL CLASS DEFINITION ==
        // == END OF EDITABLE AREA FOR ADDITIONAL CLASS DEFINITION ==
__GET_CONST_STRING__
    }
__CONST_DEFINITION__

    // == START OF EDITABLE AREA FOR BOTTOM ==
    // == END OF EDITABLE AREA FOR BOTTOM ==
}
